### PR TITLE
Story 4

### PR DESCRIPTION
**Pull request by JoelLindow**
Related Issue: #24 

Summary: Story 4 functional, may be bugs with quantity when that is implemented

Author(s): Joel, sam

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ gem 'sass-rails', '~> 5.0'
 gem 'uglifier', '>= 1.3.0'
 # See https://github.com/rails/execjs#readme for more supported runtimes
 # gem 'therubyracer', platforms: :ruby
-
+gem 'friendly_id'
```
Comment (limsammy) [inline Gemfile]: We need to remove this at some point

File: app/controllers/carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class CartsController < ApplicationController
+
+  def show
+    @items = Item.where(id: session[:cart].values)
+    # @cart = 
+  end
+
+  def create
+    id = params["item_id"].to_s
+    @item = Item.find_by(id: id)
```
Comment (limsammy) [inline app/controllers/carts_controller.rb:10]: Refactor line 10 to a helper method??

**Comment by JoelLindow** [comment]
I apologize for this premature pull request.

**Comment by JoelLindow** [comment]
By popular Demand, It's back

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,3 +1,8 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :set_cart
+
+  def set_cart
+    @cart ||= Cart.new(session[:cart])
```
Comment (amhursh) [inline app/controllers/application_controller.rb:6]: Interesting and useful. Looks good

File: app/views/categories/show.html.erb [hunk 4]
```diff
@@ -1,8 +1,8 @@
 <div>
 <%= @category.title %>
   <% @category.items.each do |item| %>
-	<div class="item_card">
-		<p><%= item.title %></p>
-	</div>
+  <div class="item_card">
```
Comment (randy-darsh) [inline app/views/categories/show.html.erb:4]: you're welcome

File: app/models/cart.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  def add_item(id)
+    contents[id.to_s] = (contents[id.to_s] || 0) + 1
+  end
+
+  def total
+    contents.map do |content|
+      item = Item.find_by(id: content.first)
+      item.price * content.last
```
Comment (amhursh) [inline app/models/cart.rb:15]: I like this. Nice and concise.

File: config/routes.rb [hunk 6]
```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
 
-	resources :items, only: [:index]
-	resources :categories, path: "/", only: [:show]
+  resources :items, only: [:index]
+
+  # resources :carts, only: [:create]
+
+  # get '/cart', to: 'carts#show', as: 'cart'
```
Comment (randy-darsh) [inline config/routes.rb:7]: remove old commented out routes

File: app/views/items/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<div>
+	<% @items.each do |item| %>
+	<div class="item_card">
+		<p><%= item.title %></p>
+		<p><%= link_to "Add to Cart", cart_path(item_id: item.id), method: :post %></p>
```
Comment (amhursh) [inline app/views/items/index.html.erb]: method post for all links like this, otherwise it defaults to get

File: app/views/items/index.html.erb [hunk 8]
```diff
@@ -2,6 +2,7 @@
 	<% @items.each do |item| %>
 	<div class="item_card">
 		<p><%= item.title %></p>
+		<p><%= link_to "Add to Cart", cart_path(item_id: item.id), method: :post %></p>
```
Comment (randy-darsh) [inline app/views/items/index.html.erb:5]: that method post

**Pull request merged**